src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  static removeDeadAnimals() {
+    Animal.alive = Animal.alive
+      .filter(animal => animal.health > 0);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
+
+  set health(value) {
+    this._health = value;
+    Animal.removeDeadAnimals();
+  }
```

Comment:
A setter is used to set a value, or to process a received value and set it to an object.
Why you use `Animal.removeDeadAnimals()` here?

Reply:
I agree, it's not necessary